Doc: Add a migration guide from Helm 2.1 to 3.0 (#2291)
grafana/mimir: docs/sources/release-notes/upgrading-helm-from-2.1-to-3.0.md

---
title: "TODO: Upgrading to mimir-distributed 3.0"
menuTitle: "TODO: Helm chart v3.0"
description: "Upgrade guide for the Grafana Mimir Helm chart version 3.0"
weight: 100
---

# Grafana Mimir Helm chart version 3.0 Upgrade Guide

## Breaking Changes

Several parameters available in version 2.1 of the mimir-distributed Helm chart have changed.
Follow these steps to migrate your Helm values file to 3.0.

- **Configuration Storage has changed**

>     - Mimir configuration is now stored in a ConfigMap by default instead of a Secret

Reviewer: ```suggestion
    - Mimir configuration is now stored in a ConfigMap by default instead of a Secret
    - In case of providing the configuration in a user managed Secret (**`useExternalConfig=true`**), you must set `configStorageType=Secret`. Conversely, it is now possible to use a ConfigMap to manage your configuration.
```

    - See [(Optional) Use the new, simplified configuration method](#optional-use-the-new-simplified-configuration-method) for tips on securing credentials in this new format
    - The previous behavior can be enabled by setting `configStorageType` to `Secret`
- **Update memcached configuration**
    - The mimir-distributed chart supports four different caches. The configuration parameters have changed as follows:

>     - If you are not overriding the value of anything under the `memcached`, `memcached-queries`, `memcached-metadata`, or `memcached-results` sections, then you can safely skip this section.

Reviewer: I think this is a bit misleading, I bet some people don't think of `enabled` as overriding stuff.

```suggestion
    - If you have not enabled any memcached caches,  and you are not overriding the value of anything under the `memcached`, `memcached-queries`, `memcached-metadata`, or `memcached-results` sections, then you can safely skip this section.
```

        - Renamed `memcached-results.replicaCount` to `results-cache.replicas`.
    - All memcached instances now share the same `ServiceAccount` that the chart uses for its services.
    - The value `memcached-*.architecture` was removed.
    - The value `memcached-*.arguments` was removed, the default arguments are now encoded in the template. Use `*-cache.extraArgs` to provide additional arguments as well as the values `*-cache.allocatedMemory`, `*-cache.maxItemMemory` and `*-cache.port` to set the memcached command line flags `-m`, `-I` and `-u`.
    - The remaining arguments are aligned with the rest of the chart's services, please consult the values file to check whether a parameter exists or was renamed.
    - **Note**: The internal service names for memcached pods have changed. If you have previously copied the value of `mimir.config` into your values file, please see the point below about updating `mimir.config` to ensure the memcached addresses are updated properly in your configuration.
- **Update serviceMonitor configuration**
    - All meta-monitoring related settings have been consolidated under the `metaMonitoring` section

>     - If you are not overriding the value of anything under the `serviceMonitor` section, then you can safely skip this section.

Reviewer: ```suggestion
    - If you have not enabled `serviceMonitor` and you are not overriding the value of anything under the `serviceMonitor` section, then you can safely skip this section.
```

    - `serviceMonitor.enabled` was renamed to `metaMonitoring.serviceMonitor.enabled`
- **Update rbac configuration**
    - PodSecurity has been made more flexible to support OpenShift deployment
    - If you are not overriding the value of anything under the `rbac` section, then you can safely skip this section.
    - `rbac.pspEnabled` has been removed
    - Use `rbac.create` along with `rbac.type` instead to select either Pod Security Policy or Security Context Constraints
    - The default behavior is unchanged
- **Update `mimir.config`**

>     - If you are not overriding the value of `mimir.config`, then you can safely skip this section.

Reviewer: ```suggestion
    - If you are not overriding the value of `mimir.config`, then you can safely skip this section.
    - Before migrating your `mimir.config` value, take a look at how to [use the new simplified configuration method](#optional-use-the-new-simplified-configuration-method).
```

    - Compare your overridden value of `mimir.config` with the one in <TODO: link>
    - In particular, pay special attention to any of the memcached addresses since these have changed.
- **Multitenancy is enabled by default**
    - All requests will now require a tenant ID is passed via the `X-Scope-OrgID` header

>     - When `enterprise.enabled` is false, the nginx gateway will transparently add a tenant id of `anonymous` when no tenant id is present in a request. This ensures compatibility with installations previously running with multitenancy disabled.

Reviewer: ```suggestion
    - When `enterprise.enabled` is false, the nginx gateway will transparently add `X-Scope-OrgId: anonymous` when no tenant id is present in a request. This ensures compatibility with installations previously running with multitenancy disabled.
```

>     - If you have overriden the nginx gateway configuration, make sure to compare it to the version in <TODO: link> to incorporate any updates.

Reviewer: ```suggestion
    - If you have overridden the nginx gateway configuration, make sure to compare it to the version in <TODO: link> to incorporate any updates.
```

        store_gateway:
          url: http://{{ template "mimir.fullname" . }}-store-gateway.{{ .Release.Namespace }}.svc:{{ include "mimir.serverHttpListenPort" . }}
    {{- end }}
    instrumentation:
      enabled: true
      distributor_client:
        address: 'dns:///{{ template "mimir.fullname" . }}-distributor-headless.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}:{{ include "mimir.serverGrpcListenPort" . }}'
    {{- end }}
```

After applying the migration steps listed in this guide, the equivalent `values.yaml` for version 3.0.0 looks like this:

First we create an external secret to store the credentials:

```yaml
apiVersion: v1
kind: Secret
metadata:
    name: mimir-bucket-secret
data:
    AWS_ACCESS_KEY_ID: FAKEACCESSKEY
    AWS_SECRET_ACCESS_KEY: FAKESECRETKEY
```

```yaml
rbac:
  create: false

chunks-cache:
  enabled: true
  replicas: 1

index-cache:
  enabled: true
  replicas: 1

metadata-cache:
  enabled: true
  replicas: 1

results-cache:
  enabled: true
  replicas: 1

metaMonitoring:
  serviceMonitor:
    enabled: true

mimir:
  structuredConfig:
    blocks_storage:
      backend: s3
      s3:

>         endpoint: s3.amazonaws.com
>         bucket_name: my-blocks-bucket
>         access_key_id: ${AWS_ACCESS_KEY_ID}
>         secret_access_key: ${AWS_SECRET_ACCESS_KEY}

Reviewer: nitpick, due to the transformation before the configuration is written to the ConfigMap, helm will put everything in alphabetic order, so to avoid confusion:

```suggestion
        access_key_id: ${AWS_ACCESS_KEY_ID}
        bucket_name: my-blocks-bucket
        endpoint: s3.amazonaws.com
        secret_access_key: ${AWS_SECRET_ACCESS_KEY}
```
